docs/developing/app-review-guide.md

```diff
@@ -0,0 +1,96 @@
+support. This can often be tested by storing some information in a grain, allowing
+Sandstorm to shut down that grain, and then opening it again and seeing if things
+persisted that an ordinary user would expect to persist between sessions.
+
+Finally, the App Market may reject apps which present security or privacy issues
+that fall outside of what a Sandstorm user would expect when using the app. This
+can mean it requests overly-broad permissions when making Powerbox requests or
+uses client-side loading to load scripts which share unexpected information outside
+of the Sandstorm sandbox. For example, we may block approval on an app which attempts
+to load an analytics script from an outside server, or does not make it suitably
+clear to a user why it needs certain outside network access prior to requesting it.
+
+**Note:** Client-side loading is a known gap in Sandstorm's security model that we
+intend to close. Not only is misuse of it a security or privacy issue, its use may
+lead to apps that break when Sandstorm closes this gap.
+
+## Commercial or proprietary apps
+
+The Sandstorm App Market permits both commercial and proprietary apps, however
+official support for app purchase and licensing is not currently available. For an
+app that is merely closed source, but free, this may not pose an issue. For apps
+which require some form of licensing to function, we expect at minimum that the
+app description clearly and prominently specifies the licensing requirements and
+activation method. We would also encourage the app to have at least a minimal level of
+functionality such as a trial or limited version, such that users can experience the
+app before deciding on a purchase. As this is a territory the Sandstorm app review
+team hasn't spent a lot of time on, interested parties may wish to reach out to the
+Sandstorm community for feedback prior to committing significant work here.
+
+## Copyright and copyleft
+
+The developer is responsible for ensuring their packages comply with all relevant
+copyright or copyleft limitations. If a package is reported to us as in violation of a
+license, it may be pulled from the App Market.
+
+Sandstorm package files (or SPKs) are an archive format which includes a variety of
+components needed to run a given application. Often the process of building an SPK pulls
+in components and libraries from open source distributions such as Debian. We believe
+these fall comfortably within the GPLv3's [system libraries exception](https://www.gnu.org/licenses/quick-guide-gplv3.html#less-source-to-distribute-new-system-libraries-exception) and falls well within the spirit of most open source licensing models.
+
+Open source Sandstorm packages generally provide all of the information necessary to modify
+and recreate them, including where to get any other components required to build the SPK,
+however, proprietary apps may need to seek legal counsel on what requirements are needed
```

Comment:
Hmm reading this paragraph I wonder if you were interpreting the system libraries exception backwards? The system libraries exception is about GPL apps built on top of proprietary platforms, not proprietary apps built on GPL platforms.

Proprietary apps definitely can't use GPL libraries at all. Most common Linux system libraries are LGPL or permissively licensed. Anyway, I'm not sure if we should be advising people on implications of GPL here...

Reply:
I don't see how the section I linked can be read the way you seem to be reading it. Python and Ruby are both open source, for instance, and have now been included in the system libraries exception. Furthermore, "The new definition also makes it clear that you can combine GPLed software with GPL-incompatible System Libraries" suggests that GPLv3 adds the clarification it can *also* be read the way you are reading it.

That being said, I think we can remove any guidance on licensing to reach a final state for this PR. Do you feel the "may need to seek legal counsel" line should be left with regards to proprietary apps on Sandstorm? Ian had suggested he wasn't sure a proprietary SPK was possible if it was grabbing from Linux packages as well.

Reply:
> Python and Ruby are both open source, for instance, and have now been included in the system libraries exception.

They chose confusing examples here. What they are saying is: "If you distribute a GPL'd Python script, you are not required to distribute the complete source code of Python, even though the GPL normally requires you to produce code for all underlying libraries you use." This has nothing to do with Python's license. Probably a better example language would have been, say, Matlab.

There is definitely no exception that allows GPL'd libraries that happen to be distributed as part of an operating system to be used by proprietary applications.

> Do you feel the "may need to seek legal counsel" line should be left with regards to proprietary apps on Sandstorm?

I think I would just say: Sandstorm app packages normally include all of an app's userspace dependencies, which typically includes many third-party libraries. App developers are responsible for complying with the licenses of all libraries that are included in their app package.

> Ian had suggested he wasn't sure a proprietary SPK was possible if it was grabbing from Linux packages as well.

All essential Linux system libraries are licensed in such a way that proprietary apps can use them. E.g. glibc is LGPL. So I don't see why a proprietary SPK would be a problem, as long as it restrict itself to LGPL or more-permissive dependencies.

The LGPL does require that proprietary apps using the library make it possible for users to relink the app against a modified version of the underlying library. This is easily satisfied by dynamic linking. A proprietary app that statically links against glibc would be problematic, though.

Reply:
>  I think I would just say: Sandstorm app packages normally include all of an app's userspace dependencies, which typically includes many third-party libraries. App developers are responsible for complying with the licenses of all libraries that are included in their app package.

This sounds good to me; I think we should basically leave it at that and avoid giving any further advice re: licensing.